### ISSUE TITLE
`$crate` is improperly resolved if it passes through another macro

### ISSUE THREAD

**Issue by jonas-schievink** [rust-analyzer/rust-analyzer#6788]
Current behavior:

```rust
//- /main.rs crate:main deps:lib
use lib::path;

macro_rules! vec {
  ( $e:expr ) => { $e };
}

fn f() {
    path![];
  //^^^^^^^ could not resolve macro `$crate::inner`
}

//- /lib.rs crate:lib
#[macro_export]
macro_rules! path {
    () => { vec![$crate::inner!()] };
}
#[macro_export]
macro_rules! inner {
    () => { 0 };
}
```

Expected behavior: The macro should be resolved correctly.

This causes incorrect diagnostics in `crates/hir_ty/src/diagnostics/expr.rs`, and probably all places that invoke the `path!` macro.

**Comment by jonas-schievink** [comment]
I think `$crate` needs to be resolved relative to the crate that contains the `$crate` tokens. I don't think we can do that in the current architecture, since we don't track hygiene of individual identifiers at the moment.

**Comment by lnicola** [comment]
Fixed in #7133.

**Comment by edwin0cheng** [comment]
Fixed in #7145 again :)
